src/main/webapp/script.js

```diff
@@ -53,37 +53,34 @@ async function getYouTubeComments(url) {
  * Wrapper function for preparing onClick function
  */
 function onButtonPress() {
-  $('.submitBtn').click(function() {
-    showLoadingGif();
-    getChart();
+  $('#submit-link-btn').click(function() {
+    const urlInput = document.getElementById('link-input').value;
+    updateUIWithVideoContext(urlInput);
+    displayOverallResults(urlInput);
   });
 }
 
 /**
  * Fetches data and adds to html
  * @param {String} url youtube url to retrieve comments
  */
-async function getChart(url) {
+async function displayOverallResults(url) {
+  showLoadingGif();
```

Comment:
Should the loading Gif be displayed before the video context has been retrieved?

Also one further improvement that we could consider (in a seperate PR) might be to have seperate loading icons for the different containers that get loaded.

And more specifically it would be cool if the user knew what was being loaded (not sure on the specific UI for this piece).